tests/x11/rrdtool_x11.pm

```diff
@@ -0,0 +1,86 @@
+# SUSE"s openQA tests
+#
+# Copyright © 2019 SUSE LLC
+#
+# Copying and distribution of this file, with or without modification,
+# are permitted in any medium without royalty provided the copyright
+# notice and this notice are preserved.  This file is offered as-is,
+# without any warranty.
+
+# Summary: Regression test for rrdtool
+# - created, update rrd file
+# - fetch data from file
+# - generate images with different options
+# - compare images with needles
+# Expected result:
+# 4 graphics will be displayed without error.
+#
+# Maintainer: Marcelo Martins <mmartins@suse.cz>
+
+use base "x11test";
+use strict;
+use warnings;
+use testapi;
+use utils;
+use version_utils 'is_sle';
+
+sub run {
+    select_console 'x11';
+    x11_start_program('xterm');
+    become_root;
+    pkcon_quit;
+    # create a tmp dir/files to work
+    assert_script_run "mkdir /tmp/rrdtool; cd /tmp/rrdtool";
+    # install requirements
+    zypper_call "in rrdtool";
+    # create a rrd file
+    assert_script_run "rrdtool create test.rrd --start 920804400  DS:speed:COUNTER:600:U:U RRA:AVERAGE:0.5:1:24 RRA:AVERAGE:0.5:6:10";
+    # update the rrd file
+    assert_script_run "rrdtool update test.rrd 920804700:12345 920805000:12357 920805300:12363 ; /
+rrdtool update test.rrd 920805600:12363 920805900:12363 920806200:12373 ; /
+rrdtool update test.rrd 920806500:12383 920806800:12393 920807100:12399 ; /
+rrdtool update test.rrd 920807400:12405 920807700:12411 920808000:12415 ; /
+rrdtool update test.rrd 920808300:12420 920808600:12422 920808900:12423";
+    # fetch the rrd file
+    assert_script_run "rrdtool fetch test.rrd AVERAGE --start 920804400 --end 920809200";
+
+    # make the graph 1, open image and verify if correct.
+    assert_script_run "rrdtool graph speed.png --start 920804400 --end 920808000 DEF:myspeed=test.rrd:speed:AVERAGE LINE2:myspeed#FF0000";
+    send_key("alt-f2");
+    type_string("eog /tmp/rrdtool/speed.png\n");
```

Comment:
You just removed call to function completely and you are repeating same steps, several times.

Reply:
ok, I think solve now.